### PR TITLE
Added support for blank placeholder spaces.

### PR DESCRIPTION
**Pull request by xrolfex**
This PR adds support for a new object type of placeholder. It offers a way to put in blank spaces to allow for custom layouts.

This is one way to address #313. It still needs localization and is a first attempt. Im not sure if this is how you would like to handle it.

image

image

image

image

image

image

### CODE REVIEW

**Comment by matt8707** [comment]
Looks good. It's somewhat related to sidebar > divider > empty option. So, it would be nice to maintain consistency in appearance. However, this could lead to confusion with "new object". So maybe that style but gray in color, what do you think?

divider

**Comment by xrolfex** [comment]
So something more like this? Are you also thinking remove the placeholder text as well?

image

**Comment by matt8707** [comment]
Yeah I'm thinking,
* Make it transparent white to differentiate from the yellow
* Use $lang('empty') instead of 'placeholder' (unsure if needed)
* Internally, rename it to 'empty'

**Comment by xrolfex** [comment]
> Yeah I'm thinking,
> 
> * Make it transparent white to differentiate from the yellow
> * Use $lang('empty') instead of 'placeholder' (unsure if needed)
> * Internally, rename it to 'empty'

Got these taken card of; did remove the text during editing:

 
image

File: src/lib/Main/Empty.svelte [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		outline-offset: -2px;
+	}
+
+	/* Phone and Tablet (portrait) */
+	@media all and (max-width: 768px) {
+		.container {
+			width: calc(50vw - 1.45rem);
+		}
+	}
+</style>
```
Comment (matt8707) [inline src/lib/Main/Empty.svelte:101]:
```suggestion
<script lang="ts">
	import { editMode, itemHeight, motion } from '$lib/Stores';
	import { openModal } from 'svelte-modals';

	export let sel: any;

	/**
	 * Handle click
	 */
	async function handleClick() {
		if ($editMode) {
			openModal(() => import('$lib/Modal/EmptyConfig.svelte'), { sel });
		}
	}
</script>

<!-- svelte-ignore a11y-click-events-have-key-events -->
<!-- svelte-ignore a11y-no-static-element-interactions -->
<div
	on:click={handleClick}
	class="container"
	style:height="{$itemHeight}px"
	style:opacity={$editMode ? '1' : '0'}
	style:transition="opacity {$motion}ms ease"
/>

<style>
	.container {
		width: 14.5rem;
		background-color: rgba(255, 255, 255, 0.25);
		border-radius: 0.65rem;
		margin: 0;
		outline: 2px dashed #fff;
		outline-offset: -2px;
	}

	/* Phone and Tablet (portrait) */
	@media all and (max-width: 768px) {
		.container {
			width: calc(50vw - 1.45rem);
		}
	}
</style>

```

File: src/lib/Modal/EmptyConfig.svelte [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		<div style:pointer-events="none">
+			<Empty {sel} />
+		</div>
+
+		<ConfigButtons {sel} />
+	</Modal>
+{/if}
+
+<style>
+</style>
```
Comment (matt8707) [inline src/lib/Modal/EmptyConfig.svelte:29]:
```suggestion
<script lang="ts">
	import { lang } from '$lib/Stores';
	import type { EmptyItem } from '$lib/Types';
	import Modal from '$lib/Modal/Index.svelte';
	import ConfigButtons from '$lib/Modal/ConfigButtons.svelte';
	import Empty from '$lib/Main/Empty.svelte';

	export let isOpen: boolean;
	export let sel: EmptyItem;
</script>

{#if isOpen}
	<Modal>
		<h1 slot="title">{$lang('empty')}</h1>
		<h2>{$lang('preview')}</h2>

		<div>
			<Empty {sel} />
		</div>

		<ConfigButtons {sel} />
	</Modal>
{/if}

<style>
	div {
		pointer-events: none;
	}
</style>

```

**Review by matt8707 — CHANGES_REQUESTED** [review]

**Comment by xrolfex** [comment]
Requested changes applied.

**Review by matt8707 — APPROVED** [review]

**Pull request merged**